### PR TITLE
✨ Adding line numbers for rest of Token-Permissions (and by extension, Packaging)

### PR DESCRIPTION
**Pull request by chrismcgehee**
Packaging)

* **Please check if the PR fulfills these requirements**
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] PR title follows the guidelines defined in  https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Feature

* **What is the current behavior?** (You can also link to an open issue here)
When the Token-Permission check makes logs for packaging workflow, it does not set the line number. Same thing for the Packaging check.
Closes https://github.com/ossf/scorecard/issues/886

* **What is the new behavior (if this is a feature change)?**
All logs from the Token-Permission and Packaging checks will include line numbers. This PR also makes the check for packaging workflow more precise. Instead of doing a string search for an action name, it will parse the workflow file and check that a `uses` field on a step matches the action name.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No

* **Other information**:

### CODE REVIEW

File: checks/testdata/github-workflow-permissions-run-package-workflow-write.yaml [hunk 1]
```diff
@@ -22,5 +22,4 @@ jobs:
       packages: write
     steps:
       - name: some name
-        run: echo "write-and-read workflow"
```
Comment (chrismcgehee) [inline checks/testdata/github-workflow-permissions-run-package-workflow-write.yaml:25]: Removed this line because parsing this file caused an error. A step is not allowed to have both a `run` and a `uses` field.

Comment (azeemshaikh38) [reply checks/testdata/github-workflow-permissions-run-package-workflow-write.yaml:25]: Nit suggestion: this is a good corner case to test for. We should probably have a test which tests that we handle these parsing issues gracefully.

**Review by azeemshaikh38 — APPROVED** [review]
Thanks @chrismcgehee! Great PR as always.

**Comment by laurentsimon** [comment]
Thanks @chrismcgehee , this is really cool!

I think this PR closes https://github.com/ossf/scorecard/issues/1100, is that right?

closes https://github.com/ossf/scorecard/issues/1100

**Comment by chrismcgehee** [comment]
> I think this PR closes #1100, is that right?

Maybe it does; I'll comment on that issue.

**Comment by chrismcgehee** [comment]
> Nit suggestion: this is a good corner case to test for. We should probably have a test which tests that we handle these parsing issues gracefully.

I'm all for testing corner cases, but I don't think this case is a good fit for our existing unit tests. If a workflow file does not conform to Github's specifications for a workflow file, I think it should be indeterminate what our code does; in this case it will still run the token-permissions check on this file, but it would treat is as a `run` step and ignore the `uses`. I think that behavior is fine, and I also think it would be fine if our code reported an error on this file. What we need to do is create a larger test that checks multiple files and continues checking if 1 of the files has some sort of error.
